src/utils/certificates.test.ts

```diff
@@ -0,0 +1,215 @@
+  storeCertificatesSpy = jest.spyOn(certificates, 'storeCertificates')
+  generateKeyPairSpy = jest.spyOn(forge.pki.rsa, 'generateKeyPair')
+  createCertificateSpy = jest.spyOn(forge.pki, 'createCertificate')
+  getPublicKeyFingerprintSpy = jest.spyOn(forge.pki, 'getPublicKeyFingerprint')
+  sha384CreateSpy = jest.spyOn(forge.md.sha384, 'create')
+})
+
+afterEach(() => {
+  jest.clearAllMocks()
+  jest.resetModules()
+})
+
+describe('constructor', () => {
+  it('should construct a Certificates object', () => {
+    const result = certificates
+    expect(result).toBeTruthy()
+  })
+})
+
+describe('generateCertificates', () => {
+  it('should generate certificates', () => {
+    const generateRootCertificateSpy = jest.spyOn(certificates, 'GenerateRootCertificate').mockReturnValue({})
+    const certificateToPemSpy = jest.spyOn(forge.pki, 'certificateToPem').mockReturnValue('certificate')
+    const privateKeyToPemSpy = jest.spyOn(forge.pki, 'privateKeyToPem').mockReturnValue('private key')
+    const issueWebServerCertificateSpy = jest.spyOn(certificates, 'IssueWebServerCertificate').mockReturnValue({})
+    const result: certificatesType = certificates.generateCertificates()
+    expect(result.mps_tls_config).toBeTruthy()
+    expect(result.web_tls_config).toBeTruthy()
+    expect(result.root_key).toBeTruthy()
+    expect(certificateToPemSpy).toBeCalled()
+    expect(privateKeyToPemSpy).toBeCalled()
+    expect(issueWebServerCertificateSpy).toBeCalled()
+    expect(generateRootCertificateSpy).toBeCalled()
+  })
+})
+
+describe('GenerateRootCertificate', () => {
+  const keyPair = {
+    publicKey: {},
+    privateKey: {}
+  } as any
+  const certificate = {
+    validity: {},
+    setSubject: jest.fn(),
+    setIssuer: jest.fn(),
+    setExtensions: jest.fn(),
+    sign: jest.fn()
+  } as any
+
+  beforeEach(() => {
+    generateKeyPairSpy.mockReturnValue(keyPair)
+    createCertificateSpy.mockReturnValue(certificate)
+    getPublicKeyFingerprintSpy.mockReturnValue('abc123')
+    sha384CreateSpy.mockReturnValue(123456 as any)
+  })
+
+  it('should generate root certificate using strong keysize', () => {
+    const addThumbPrintToName = true
+    const commonName = 'cn'
+    const country = 'us'
+    const organization = 'rhbe'
+    const strong = true
+    const result = certificates.GenerateRootCertificate(addThumbPrintToName, commonName, country, organization, strong)
+    expect(result.cert).toBeTruthy()
+    expect(result.key).toBeTruthy()
+  })
+
+  it('should generate root certificate without using strong keysize', () => {
+    const addThumbPrintToName = true
+    const commonName = null
+    const country = null
+    const organization = null
+    const strong = false
+    const result = certificates.GenerateRootCertificate(addThumbPrintToName, commonName, country, organization, strong)
+    expect(result.cert).toBeTruthy()
+    expect(result.key).toBeTruthy()
+  })
+})
+
+describe('GenerateRootCertificate', () => {
```

Comment:
Should this describe be 'getCertificates' instead?

Reply:
Yup, good catch! This is addressed in 7e85f6ff07c1b3ee35ef4a6ca8dbb078423b4ccd